### ISSUE TITLE
Nested forwarded types (Outer+Inner) are never recognized as forwarders

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3480]
Found in adversarial review of #3419 (PR #3476).  `ForwardedTypeAliases.ReadForwarder` accepts a forwarder row only when its implementation is a direct `AssemblyReference`, and compares a flat `Namespace.Name`:  ```csharp if (exported.Implementation.Kind != HandleKind.AssemblyReference)     continue; ```  A **nested** forwarded type does not look like that. Its `ExportedType` row's implementation is *another `ExportedType`* (the enclosing type), and its identity is `Outer+Inner`. So nested forwarded types are never recognized and calls through them are dropped.  ## Reproduction  ```powershell dotnet-inspect member "Outer.Inner" Ping --library Definer.dll -S Callers --bin <scope> ```  ```text No callers found in this assembly. ```  Running the same caller against the forwarding facade returns `1` from the target implementation, proving it is a real caller.  ## Direction of the failure  A **conservative miss** — the alias is never recorded, so nothing false is reported. Same safe direction as #3479, and the reason it is Important rather than Blocking.  ## Suggested approach  Walk the nested `ExportedType` parent chain, reconstruct the `Outer+Inner` identity, and inherit the outermost row's target assembly reference. Needs a matching change wherever the target type name is spelled for comparison, since the current comparison builds `Namespace.Name` only.
